src/Environments/MacOS/ResourceFork.cs

```diff
@@ -342,7 +343,115 @@ public void Dump()
+                                    int a5rl1 = 0;
+                                    int a5rl2 = 0;
+                                    int a5rpt = 0;
+                                    var a5copylength = 0;
+                                    bool a5dend = false;
+
+                                    do
+                                    {
+                                        a5rpt = 1;
+                                        a5rl1 = a5dr.ReadByte();
+                                        a5rl2 = a5rl1;
+                                        a5rl1 = a5rl1 & 0x0F;
+                                        if (a5rl1 == 0)
+                                        {
+                                            a5rl1 = GetRL(a5dr, ref a5rpt);
+                                            if (a5rl1 == 0)
+                                            {
+                                                a5dend = true;
+                                            }
+
+                                        }
+                                        else
+                                        {
+                                            a5rl1 = a5rl1 * 2;
+                                        }
+                                        if (!a5dend)
+                                        {
+                                            a5rl2 = a5rl2 & 0xF0;
+                                            if (a5rl2 == 0)
+                                            {
+                                                a5rl2 = GetRL(a5dr, ref a5rpt);
+                                            }
+                                            else
+                                            {
+                                                a5rl2 = a5rl2 >> 3;
+                                            }
+                                            do
+                                            {
+                                                a5beloww.Position = a5beloww.Position + a5rl2;
+                                                a5copylength = a5rl1;
+                                                do
+                                                {
+                                                    a5beloww.WriteByte(a5dr.ReadByte());
+                                                    a5copylength -= 1;
+                                                } while (a5copylength > 0);
+                                                a5rpt -= 1;
+                                            } while (a5rpt > 0);
+                                        }
+                                    } while (!a5dend);
+                                }
+                            }
+                        }
+                    }
+                }
+                    
+            }
+        }
+
+        private int GetRL(BeImageReader a5dr, ref int a5rpt)
+        {
+            int RL = a5dr.ReadByte();
```

Comment:
Consider adhering to the coding style: local variables are in camel case. We can afford a longer, more self-explaining name: `runLength`.